packages/rum-core/src/performance-monitoring/breakdown.js

```diff
@@ -41,12 +41,59 @@ function getValue(value) {
   return { value }
 }
 
+function calculateSelfTime(transaction) {
+  const { spans, _start, _end } = transaction
+  /**
+   * When there are no spans transaction duration accounts for
+   * the overall self time
+   */
+  if (spans.length === 0) {
+    return transaction.duration()
+  }
+  /**
+   * The below selfTime calculation logic assumes the spans are sorted
+   * based on the start time
+   */
+  spans.sort((span1, span2) => span1._start - span2._start)
+
+  let span = spans[0]
+  let spanEnd = span._end
+  let spanStart = span._start
+
+  let lastContinuousEnd = spanEnd
+  let selfTime = spanStart - _start
```

Comment:
Could this be negative? We moved the logic for adjusting the transaction times to the earliest and latest span to `PerformanceMonitoring.adjustTransactionTime` but at that point we have already calculated breakdowns.

Maybe we should refactor `adjustTransactionTime` to run after `Transaction.end` again.
We also need test for this case

Reply:
Done